sdk/src/main/scala/com/horizen/account/sc2sc/AbstractCrossChainRedeemMessageProcessor.scala

```diff
@@ -87,11 +89,11 @@ abstract class AbstractCrossChainRedeemMessageProcessor(
                                             scCommitmentTreeRoot: Array[Byte],
                                             nextScCommitmentTreeRoot: Array[Byte],
                                             view: BaseAccountStateView): Unit = {
-    if (!view.doesScTxCommitmentTreeRootExist(scCommitmentTreeRoot)) {
+    if (view.getAccountStorage(contractAddress, scCommitmentTreeRoot).isEmpty) {
```

Comment:
the treeroot is not saved in the account storage.
must be taken from the metadatastorage (throug the accountstateview that is the implementation of the BaseAccountStateView hierarchy)